### PR TITLE
Adds proc to return list of fields active on a turf

### PR DESCRIPTION
**Pull request by silicons**
going to need this soon (tm)

### CODE REVIEW

File: code/modules/fields/turf_objects.dm [hunk 1]
```diff
@@ -75,3 +75,10 @@
 		if(O.parent == F)
 			return FIELD_TURF
 	return NO_FIELD
+
+/proc/return_fields_on_turf(turf/T)	//Only works if a field sets up turf objects on all turfs.
+	var/list/ret = list()
+	fro(var/obj/effect/abstract/proximity_checker/advanced/part in T)
```
Comment (RemieRichards) [inline code/modules/fields/turf_objects.dm]: fro fro, fight the power!

Comment (kevinz000) [inline code/modules/fields/turf_objects.dm]: aAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by lzimann** [comment]
Why add a proc that is not being used? Even if you will it's a simple thing, you can easily add it together with whatever you want to do.

**Comment by kevinz000** [comment]
@lzimann mostly so I don't have to change that many files in one PR so it's easier to review I guess.

File: code/modules/fields/turf_objects.dm [hunk 2]
```diff
@@ -75,3 +75,10 @@
 		if(O.parent == F)
 			return FIELD_TURF
 	return NO_FIELD
+
+/proc/return_fields_on_turf(turf/T)	//Only works if a field sets up turf objects on all turfs.
+	var/list/ret = list()
+	for(var/obj/effect/abstract/proximity_checker/advanced/part in T)
+		if(part.parent)
+			ret |= part.parent
```
Comment (Cyberboss) [inline code/modules/fields/turf_objects.dm]: Do ret[part.parent] = TRUE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Jordie0608** [comment]
One more file changed on a PR doesn't sound like a hassle.

**Comment by Really-Good-Soda-Flavor** [comment]
Please just put this in whatever pull request will _actually use it_.

**Comment by lzimann** [comment]
Just add this whenever you are actually using it.

**Pull request closed without merging**